src/app/project/project-documents/detail/detail.component.ts

```diff
@@ -19,7 +20,7 @@ import { DocumentSection } from 'app/models/documentSection';
 })
 export class DocumentDetailComponent implements OnInit, OnDestroy {
   private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
-  public document: Document = null;
+  public document = null; // Document or ExternalLink
```

Comment:
Could you use a union type here to keep type safety? 

```suggestion
  public document: Document | ExternalLink = null; // Document or ExternalLink
```

Reply:
Unfortunately not without refactoring code and making it more verbose, as mentioned previously. The typing was done to the degree that typing has been performed elsewhere in the codebase.